compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -631,7 +631,7 @@ class SpaceEngine(using Context) extends SpaceLogic {
       case tp =>
         def getChildren(sym: Symbol): List[Symbol] =
           sym.children.flatMap { child =>
-            if child eq sym then Nil // i3145: sealed trait Baz, val x = new Baz {}, Baz.children returns Baz...
+            if child eq sym then List(sym) // i3145: sealed trait Baz, val x = new Baz {}, Baz.children returns Baz...
```

Comment:
`sealed trait Baz, val x = new Baz {}, Baz.children returns Baz`

Which exactly `Baz` in `Baz.children` are we talking about here?